clusterctl/cmd/validate_cluster.go

```diff
@@ -0,0 +1,50 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"github.com/golang/glog"
+	"github.com/spf13/cobra"
+	"k8s.io/kubernetes/pkg/kubectl/util/i18n"
+	"sigs.k8s.io/cluster-api/errors"
+)
+
+type ValidateClusterOptions struct {
+	Name string
+}
+
+var vco = &ValidateClusterOptions{}
+
+var validateClusterCmd = &cobra.Command{
+	Use:   "cluster",
+	Short: i18n.T("Validate a cluster created by cluster API."),
+	Long:  i18n.T("Validate a cluster created by cluster API."),
+	Run: func(cmd *cobra.Command, args []string) {
+		if err := RunValidateCluster(vco); err != nil {
+			glog.Exit(err)
+		}
+	},
+}
+
+func RunValidateCluster(vco *ValidateClusterOptions) error {
+	return errors.NotImplementedError
+}
+
+func init() {
+	validateClusterCmd.Flags().StringVarP(&vco.Name, "name", "n", "", "Cluster name")
```

Comment:
Please see jessica's comment here: https://github.com/kubernetes-sigs/cluster-api/pull/119#discussion_r185310852

> Please leave all flags out of the skeleton. They can be filled in as code uses them. This ensures the the flags actually do something when they appear and also ensure they are defined in a way that works and is accurate to the behavior of the code.

Reply:
OK. I just removed it. Will add them in follow-up PRs.